### PR TITLE
fix: correctly compare versions to find the latest version on app hub [DHIS2-10859]

### PR DESCRIPTION
**Pull request by amcgee**
Fixes https://jira.dhis2.org/browse/DHIS2-10859

There was a bug in version comparisons which caused the wrong version to be picked if multiple existed on the app hub.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "app-management-app",
-    "version": "31.0.0",
+    "version": "31.2.0",
```
Comment (Birkbjo) [inline package.json:3]: Any reason for two minor bumps?

**Review by Birkbjo — DISMISSED** [review]
LGTM!

File: package.json (hunk 1 above)
Comment (amcgee) [reply package.json:3]: Just pushed twice to the App Hub... though I deleted the older versions.  We're going to want to have some more sane versioning at some point soon so likely not a big deal, but I can change it to 31.1.0 if you want

**Review by Birkbjo — APPROVED** [review]

**Comment by amcgee** [comment]
This didn't get into 2.35 for some reason (looks like it never got RCB approval), I think it's critical to fix in 2.36.0 and backport to 2.35 as well.

**Comment by amcgee** [comment]
@Birkbjo @mediremi I rebased this on master, will then backport to 2.36.0, v36, and v35

**Review by mediremi — APPROVED** [review]

**Review by Birkbjo — APPROVED** [review]

**Pull request merged**